chain/txValidator.go

```diff
@@ -421,6 +472,24 @@ func (bvs *BlockValidationState) VerifyTransactionWithBlock(txn *transaction.Tra
 		}
 		bvs.generateIDs[publicKey] = struct{}{}
 		amount = Fixed64(generateIdPayload.RegistrationFee)
+	case pb.NanoPayType:
+		npPayload := payload.(*pb.NanoPay)
+		if npPayload.Height < header.UnsignedHeader.Height {
+			return errors.New("[VerifyTransactionWithBlock], nano pay expired")
+		}
+		key := nanoPay{BytesToHexString(npPayload.Sender), BytesToHexString(npPayload.Recipient), npPayload.Nonce}
```

Comment:
why not just using getNanoPayId to compute key?

Reply:
I guess because current key doesn't need to serialize anything, so it should be faster.

Reply:
Haha, then what about switching to getNanoPayId to make it less likely to have bugs?

Reply:
I'm not sure they're equivalent, e.g. if nano pay state were to be moved into sender's state (for whatever reason), then getNanoPayId wouldn't use sender, but key in validator still must use it.

Reply:
I see, that makes sense